src/bsoncxx/json.cpp

```diff
@@ -274,15 +274,16 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
-#define BSONCXX_ENUM(name, val)            \
-    case val:                              \
-        v.visit_value(value.get_##name()); \
-        break;
+    if(value.type()) {
```

Comment:
For now, do you think I should catch the exception in to_json(document::view
view)?

On Mon, Dec 7, 2015 at 11:47 AM, Samantha Ritter <notifications@github.com>
wrote:

> In src/bsoncxx/json.cpp
> <https://github.com/mongodb/mongo-cxx-driver/pull/387#discussion_r46844121>
> :
>
> > @@ -274,15 +274,16 @@ std::string to_json(types::value value) {
> >
> >      json_visitor v(ss, false, 0);
> >
> > -    switch ((int)value.type()) {
> > -#define BSONCXX_ENUM(name, val)            \
> > -    case val:                              \
> > -        v.visit_value(value.get_##name()); \
> > -        break;
> > +    if(value.type()) {
>
> You could default-construct a document::view and attempt to use it, then
> it would be invalid in a similar way.
>
> I think that we have a larger decision to make overall here in how
> default-constructed "invalid" types (like this one, document::view,
> uninitialized clients, etc.) should behave. I think all such types in the
> driver will currently just seg fault if you use them this way, but we have
> open tickets like CXX-742 <https://jira.mongodb.org/browse/CXX-742>
> requesting that we fail in a friendlier way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mongodb/mongo-cxx-driver/pull/387/files#r46844121>.
>



-- 

*{* name     : "Tess Avitabile",
  title    : "Software Engineer 2",
  phone    : "518-533-8094",
  location : "New York, NY",
  twitter  : "@MongoDB
<https://www.google.com/url?q=https%3A%2F%2Ftwitter.com%2Fmongodb&sa=D&sntz=1&usg=AFQjCNGEAIAhZyZhF7Z9ORWsRliTuc-2dg>
",
  facebook : "MongoDB
<https://www.google.com/url?q=https%3A%2F%2Fwww.facebook.com%2Fmongodb&sa=D&sntz=1&usg=AFQjCNGPMcaFBzmWsh-MpaWeTH6vMQoDIg>
" *}*